app/src/androidTest/java/ch/hikemate/app/ui/map/HikeDetailsScreenTest.kt

```diff
@@ -224,6 +224,10 @@ class HikeDetailScreenTest {
     doNothing().`when`(mockNavigationActions).goBack()
 
     composeTestRule.onNodeWithTag(BACK_BUTTON_TEST_TAG).performClick()
+
+    composeTestRule.waitForIdle()
+
+    assert(listOfHikeRoutesViewModel.selectedHikeRoute.value == null)
```

Comment:
The test was failing because he wasn't waiting for the newly created LaunchedEffect to happen. This just fix it and add a small check to verify the goBack is done correctly!